### PR TITLE
instead of keeping a copy of the version map in the handle, use the s…

### PR DESCRIPTION
**Pull request by galganif**
…torage proxy one, mirrors #4741 (issue #4712)

### CODE REVIEW

File: java/arcs/core/storage/handle/CollectionImpl.kt [hunk 1]
```diff
@@ -61,8 +61,8 @@ class CollectionImpl<T : Referencable>(
      */
     suspend fun store(entity: T): Boolean {
         log.debug { "Storing: $entity" }
-        versionMap.increment()
-        return storageProxy.applyOp(CrdtSet.Operation.Add(name, versionMap, entity))
+        return storageProxy.applyOp(
+            CrdtSet.Operation.Add(name, versionMap().increment(), entity))
```
Comment (jasonwyatt) [inline java/arcs/core/storage/handle/CollectionImpl.kt]: Wrapping long calls in kotlin should look like this:
```kotlin
return storageProxy.applyOp(
    CrdtSet.Operation.Add(name, versionMap().increment(), entity)
)
```

File: java/arcs/core/storage/handle/CollectionImpl.kt [hunk 2]
```diff
@@ -77,8 +77,9 @@ class CollectionImpl<T : Referencable>(
      */
     suspend fun clear(): Boolean {
         log.debug { "Clearing" }
-        return storageProxy.getParticleView().value.all {
-            storageProxy.applyOp(CrdtSet.Operation.Remove(name, versionMap, it))
+        return storageProxy.getParticleView().all {
+            storageProxy.applyOp(
+                CrdtSet.Operation.Remove(name, versionMap(), it))
```
Comment (jasonwyatt) [inline java/arcs/core/storage/handle/CollectionImpl.kt]: Same here

File: java/arcs/core/storage/handle/CollectionImpl.kt [hunk 3]
```diff
@@ -93,6 +94,7 @@ class CollectionImpl<T : Referencable>(
      */
     suspend fun remove(entity: T): Boolean {
         log.debug { "Removing $entity" }
-        return storageProxy.applyOp(CrdtSet.Operation.Remove(name, versionMap, entity))
+        return storageProxy.applyOp(
+            CrdtSet.Operation.Remove(name, versionMap(), entity))
```
Comment (jasonwyatt) [inline java/arcs/core/storage/handle/CollectionImpl.kt]: And here

File: java/arcs/core/storage/handle/SingletonImpl.kt [hunk 4]
```diff
@@ -48,8 +48,9 @@ class SingletonImpl<T : Referencable>(
      * did not apply fully. Fetch the latest value and retry.
      * */
     suspend fun store(entity: T): Boolean {
-        versionMap.increment()
-        return storageProxy.applyOp(CrdtSingleton.Operation.Update(name, versionMap, entity))
+        return storageProxy.applyOp(
+            CrdtSingleton.Operation.Update(
+                name, versionMap().increment(), entity))
```
Comment (jasonwyatt) [inline java/arcs/core/storage/handle/SingletonImpl.kt]: And here

```kotlin
return storageProxy.applyOp(
    CrdtSingleton.Operation.Update(
        name,
        versionMap().increment(),
        entity
    )
)
```

Pro-tip, you can do this with Alt+Enter while your cursor is on one of the arguments, and choose "Place arguments on separate lines"

File: java/arcs/core/storage/handle/SingletonImpl.kt [hunk 5]
```diff
@@ -63,6 +64,7 @@ class SingletonImpl<T : Referencable>(
         // Sync before clearing in order to get an updated versionMap. This ensures we can clear
         // values set by other actors.
         fetch()
-        return storageProxy.applyOp(CrdtSingleton.Operation.Clear(name, versionMap))
+        return storageProxy.applyOp(
+            CrdtSingleton.Operation.Clear(name, versionMap()))
```
Comment (jasonwyatt) [inline java/arcs/core/storage/handle/SingletonImpl.kt]: Here too

File: java/arcs/core/storage/handle/SingletonImpl.kt (hunk 4 above)
Comment (galganif) [reply java/arcs/core/storage/handle/SingletonImpl.kt]: Thanks, all done.

**Pull request merged**